tests/test_date_parser.py

```diff
@@ -276,6 +276,81 @@ def test_dates_without_preference(self, date_string, expected):
+        param('April 2015', today=datetime(2015, 2, 28), expected=datetime(2015, 4, 28)),
+        param('December 2014', today=datetime(2015, 2, 15), expected=datetime(2014, 12, 15)),
+    ])
+    def test_dates_with_day_missing_prefering_current_day_of_month(self, date_string, today=None, expected=None):
+        self.given_configuration('PREFER_DAY_OF_MONTH', 'current')
+        self.given_utcnow(today)
+        self.given_parser()
+        self.given_date_string(date_string)
+        self.when_date_is_parsed()
+        self.then_date_was_parsed_by_date_parser()
+        self.then_date_obj_exactly_is(expected)
+
+    @parameterized.expand([
+        param('February 2015', today=datetime(2015, 1, 1), expected=datetime(2015, 2, 28)),
+        param('February 2012', today=datetime(2015, 1, 1), expected=datetime(2012, 2, 29)),
+        param('March 2015', today=datetime(2015, 1, 25), expected=datetime(2015, 3, 31)),
+        param('April 2015', today=datetime(2015, 1, 15), expected=datetime(2015, 4, 30)),
+        param('April 2015', today=datetime(2015, 2, 28), expected=datetime(2015, 4, 30)),
+        param('December 2014', today=datetime(2015, 2, 15), expected=datetime(2014, 12, 31)),
+    ])
+    def test_dates_with_day_missing_prefering_last_day_of_month(self, date_string, today=None, expected=None):
+        self.given_configuration('PREFER_DAY_OF_MONTH', 'last')
+        self.given_utcnow(today)
+        self.given_parser()
+        self.given_date_string(date_string)
+        self.when_date_is_parsed()
+        self.then_date_was_parsed_by_date_parser()
+        self.then_date_obj_exactly_is(expected)
+
+    @parameterized.expand([
+        param('February 2015', today=datetime(2015, 1, 8), expected=datetime(2015, 2, 1)),
+        param('February 2012', today=datetime(2015, 1, 7), expected=datetime(2012, 2, 1)),
+        param('March 2015', today=datetime(2015, 1, 25), expected=datetime(2015, 3, 1)),
+        param('April 2015', today=datetime(2015, 1, 15), expected=datetime(2015, 4, 1)),
+        param('April 2015', today=datetime(2015, 2, 28), expected=datetime(2015, 4, 1)),
+        param('December 2014', today=datetime(2015, 2, 15), expected=datetime(2014, 12, 1)),
+    ])
+    def test_dates_with_day_missing_prefering_first_day_of_month(self, date_string, today=None, expected=None):
+        self.given_configuration('PREFER_DAY_OF_MONTH', 'first')
+        self.given_utcnow(today)
+        self.given_parser()
+        self.given_date_string(date_string)
+        self.when_date_is_parsed()
+        self.then_date_was_parsed_by_date_parser()
+        self.then_date_obj_exactly_is(expected)
+
+    @parameterized.expand([
+        param('14 February 2015', prefer_day_of_month='current', expected=datetime(2015, 2, 14)),
+        param('1 January 2015', prefer_day_of_month='last', expected=datetime(2015, 1, 1)),
+        param('30 April 2015', prefer_day_of_month='first', expected=datetime(2015, 4, 30)),
+    ])
```

Comment:
Hm, it looks like i was hasty to suggest three attributes for the test. I think we can get same result only iterating via `prefer_day_of_month`, and fixing f.ex. `date_string='14 February 2015', today='22 February 2015'` inside the test.